toolkit/tools/pkg/imagecustomizerlib/extractpartitions.go

```diff
@@ -79,11 +118,165 @@ func compressWithZstd(partitionRawFilepath string) (partitionFilepath string, er
 		return "", fmt.Errorf("failed to compress %s with zstd:\n%w", partitionRawFilepath, err)
 	}
 
-	// Remove raw file since output partition format is raw-zst.
-	err = os.Remove(partitionRawFilepath)
+	return partitionRawFilepath + ".zst", nil
+}
+
+// Prepend a skippable frame with the metadata payload to the specified partition file.
+func addSkippableFrame(partitionFilepath string, magicNumber uint32, frameSize uint32, payload [4]uint32) (err error) {
```

Comment:
This payload size of 4 is also a good candidate for a global constant. It is reused many times in the newly added code. Is it somehow determined by the framesize?

Reply:
Updated the code and it no longer makes use of this `payload` variable and also all the sizes are now defined based on the constants so none are hardcoded anymore👍.